pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -258,12 +258,13 @@ func Newk8sOrchestrator(ctx context.Context, controllerClusterFlavor cnstypes.Cn
 
 			if (controllerClusterFlavor == cnstypes.CnsClusterFlavorWorkload &&
 				k8sOrchestratorInstance.IsFSSEnabled(ctx, common.FakeAttach)) ||
-				(controllerClusterFlavor == cnstypes.CnsClusterFlavorVanilla) {
-				initVolumeHandleToPvcMap(ctx)
+				(controllerClusterFlavor == cnstypes.CnsClusterFlavorVanilla &&
+					k8sOrchestratorInstance.IsFSSEnabled(ctx, common.ListVolumes)) {
+				initVolumeHandleToPvcMap(ctx, controllerClusterFlavor)
 			}
 
 			if k8sOrchestratorInstance.IsFSSEnabled(ctx, common.ListVolumes) {
-				initVolumeNameToNodesMap(ctx)
+				initVolumeNameToNodesMap(ctx, controllerClusterFlavor)
```

Comment:
This will be called twice for WCP.

Once when `controllerClusterFlavor == cnstypes.CnsClusterFlavorWorkload &&
				k8sOrchestratorInstance.IsFSSEnabled(ctx, common.FakeAttach)` satisfies the condition

And also when `k8sOrchestratorInstance.IsFSSEnabled(ctx, common.ListVolumes)` is set to true.

Reply:
How about this:

```
			// WCP flavor specific
			if controllerClusterFlavor == cnstypes.CnsClusterFlavorWorkload{
				if (k8sOrchestratorInstance.IsFSSEnabled(ctx, common.FakeAttach) ||
			k8sOrchestratorInstance.IsFSSEnabled(ctx, common.ListVolumes)) {
				initVolumeHandleToPvcMap(ctx, controllerClusterFlavor)
			}
			// Vanilla flavor specific
			if (controllerClusterFlavor == cnstypes.CnsClusterFlavorVanilla && 
				k8sOrchestratorInstance.IsFSSEnabled(ctx, common.ListVolumes) &&
				serviceMode != "node") {
				initVolumeNameToNodesMap(ctx, controllerClusterFlavor)
			}
			// Common for all flavors
			if k8sOrchestratorInstance.IsFSSEnabled(ctx, common.ListVolumes) {
					initNodeIDToNameMap(ctx)
			}
```

Found this to be better for code readability.

Reply:
No, the second condition is a combined one:
`(controllerClusterFlavor == cnstypes.CnsClusterFlavorVanilla && k8sOrchestratorInstance.IsFSSEnabled(ctx, common.ListVolumes))`
It will not be called twice.

Reply:
initVolumeNameToNodesMap is not Vanilla specific. It is required for both WCP and vanilla.

Reply:
I got confused with the name of the map `initVolumeHandleToPvcMap` & `initVolumeNameToNodesMap`. I misread the lines and thought they are same. The current changes in the PR looks good